File: paiza/python/DP/Leetcode/70.climbing_Stairs.py
class Solution:
  def climbStairs(self, n: int) -> int:
    dp = [1,2]
    for i in range(2,n):
      dp.append(0)
      dp[i] = dp[i-1] + dp[i-2]
    return dp[n-1]

sol = Solution()
sol.climbStairs(3)

File: Leetcode_bests100/DP/70.climbing_stairs.java
class Main{
  public static void main(String[] args){
    Solution sl = new Solution();
    System.out.println(sl.climbStairs(2));
    System.out.println(sl.climbStairs(3));


  }
}

class Solution{
  public int climbStairs(int n){
    if (n==1){
      return 1;
    }
    int[] dp = new int[n];
    dp[0] = 1;
    dp[1] = 2;
    for (int i=2; i<n; i++){
      dp[i] = dp[i-1] + dp[i-2];
    }
    return dp[dp.length-1];
  }
}